Source/Task/TaskQueue.cpp

```diff
@@ -301,36 +314,35 @@ HRESULT __stdcall TaskQueuePortImpl::QueueItem(
 {
     RETURN_IF_FAILED(VerifyNotTerminated(portContext));
 
-    std::unique_ptr<QueueEntry> entry(new (std::nothrow) QueueEntry);
-    RETURN_IF_NULL_ALLOC(entry);
+    QueueEntry entry;
+    referenced_ptr<ITaskQueuePortContext> portContextHolder(portContext);
 
-    entry->portContext = portContext;
-    entry->portContext->AddRef();
-    entry->callback = callback;
-    entry->callbackContext = callbackContext;
-    entry->waitRegistration = nullptr;
-    entry->refs = 1;
+    entry.portContext = portContext;
+    entry.callback = callback;
+    entry.callbackContext = callbackContext;
+    entry.waitRegistration = nullptr;
+    entry.id = m_nextId++;
 
     if (waitMs == 0)
     {
-        entry->enqueueTime = 0;
-        RETURN_HR_IF_FALSE(E_OUTOFMEMORY, AppendEntry(entry.get()));
+        entry.enqueueTime = 0;
+        RETURN_HR_IF(E_OUTOFMEMORY, !AppendEntry(entry));
```

Comment:
why not RETURN_HR_IF_FALSE?

Reply:
Some branches in Windows are consolidating these macros.